### PR TITLE
Implement ruling hash to appropriate complexity of Rabin Karp

### PR DESCRIPTION
**Pull request by mahbubcseju**
- Previous solution is based on the hash function of python.
- Implemented ruling hash to get the appropriate complexity of rabin karp.

### CODE REVIEW

File: strings/rabin_karp.py [hunk 1]
```diff
@@ -12,22 +17,38 @@ def rabin_karp(pattern, text):
...
+        # Calculating the ruling hash
+        text_hash = (
+            (text_hash - ord(text[i]) * modulus_power) * alphabet_size
+            + ord(text[i + p_len])
+        ) % modulus
     return False
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```
Comment (cclauss) [inline strings/rabin_karp.py]: Let get this tested by our automated testing...

Change this line to: __def test_rabin_karp():__ and then at the bottom of the file, add:
```python
if __name__ == "__main__":
    test_rabin_karp()
```

This should allow [__pytest__](https://github.com/TheAlgorithms/Python/blob/master/.travis.yml#L12) to find and execute the __test_rabin_karp()__ function on each pull request.

**Comment by cclauss** [comment]
Could you please add a test containing Unicode patterns?

**Comment by mahbubcseju** [comment]
> Could you please add a test containing Unicode patterns?

Actually I dont know about Unicode pattern testing. Trying to learn about it

**Comment by cclauss** [comment]
Pytest did not find our automated tests.  :-(
* https://travis-ci.org/TheAlgorithms/Python/builds/562961905#L669

Can you please add a docetest by changing the code to read:
```python
def test_rabin_karp():
    """
    >>> test_rabin_karp()
    Success.
    """
    [ then the body of the function as it is and then at the end of the function... ]
    print("Success.")
```

**Comment by cclauss** [comment]
You should be able to test the __doctest__ locally with:
__python3 -m doctest -v strings/rabin_karp.py__

**Comment by mahbubcseju** [comment]
> You should be able to test this **doctest** locally with:
> **python3 -m doctest -v strings/rabin_karp.py**

Yes , Done !

**Comment by mahbubcseju** [comment]
Yes, please merge.

**Pull request merged**